dds-protocol-lib/src/ConnectionManagerImpl.h

```diff
@@ -242,7 +280,7 @@ namespace dds
                             break;
                     }
                 }
-                return result;
+                return std::move(result);
```

Comment:
By using `move` here you are disabling the return value optimization. The return statement was already perfect before - compiler will construct the `result` object in the place where it is called. But by calling `move` you prevent it from doing that and force move constructor call. More info: [1](https://en.wikipedia.org/wiki/Copy_elision#Return_value_optimization), [2](https://shaharmike.com/cpp/rvo/).

Reply:
It was an unintended change as we utilise RVO a lot across the whole project.
This change is a left over after we redesigned and debugged this part of the code and its surroundings.
Thanks for noticing this. I will fix.